django/forms/models.py

```diff
@@ -271,6 +272,7 @@ def __new__(mcs, name, bases, attrs):
             return new_class
 
         opts = new_class._meta = ModelFormOptions(getattr(new_class, "Meta", None))
+        formfield_callback = opts.formfield_callback or formfield_callback
```

Comment:
Can we simply fix for ticket-24974? I don't see any tests failures with:
```diff
diff --git a/django/forms/models.py b/django/forms/models.py
index 5bd7ef256a..95bce27d9a 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -258,21 +258,12 @@ class ModelFormOptions:
 
 class ModelFormMetaclass(DeclarativeFieldsMetaclass):
     def __new__(mcs, name, bases, attrs):
-        base_formfield_callback = None
-        for b in bases:
-            if hasattr(b, "Meta") and hasattr(b.Meta, "formfield_callback"):
-                base_formfield_callback = b.Meta.formfield_callback
-                break
-
-        formfield_callback = attrs.pop("formfield_callback", base_formfield_callback)
-
         new_class = super().__new__(mcs, name, bases, attrs)
 
         if bases == (BaseModelForm,):
             return new_class
 
         opts = new_class._meta = ModelFormOptions(getattr(new_class, "Meta", None))
-        formfield_callback = opts.formfield_callback or formfield_callback
 
         # We check if a string was passed to `fields` or `exclude`,
         # which is likely to be a mistake where the user typed ('foo') instead
@@ -310,7 +301,7 @@ class ModelFormMetaclass(DeclarativeFieldsMetaclass):
                 opts.fields,
                 opts.exclude,
                 opts.widgets,
-                formfield_callback,
+                opts.formfield_callback,
                 opts.localized_fields,
                 opts.labels,
                 opts.help_texts,
```

Reply:
I was also thinking about it but that drops support for `formfield_callback` in class attributes like:

```python3
class NewForm(forms.ModelForm):
    formfield_callback = callback
    
    class Meta:
        model = Person
        widgets = {"name": widget}
        fields = ["id", "name"]
```

On the other hand, this way is not tested or documented. If it's not a problem to drop it, I am in favor of this solution.

Reply:
Does it work now? I wouldn't hesitate to simplify fix for 24974.

Reply:
Right now:
1. `formfield_callback` in `Meta`: ❌  does not work (callback never called)
```python3
class NewForm(forms.ModelForm):
    class Meta:
        model = Person
        fields = ["id", "name"]
        formfield_callback = callback
```

2. `formfield_callback` as a class attribute: ✅ works (callback called)
```python3
class NewForm(forms.ModelForm):
    formfield_callback = callback
    class Meta:
        model = Person
        fields = ["id", "name"]
```

3. Inheriting from a form with `formfield_callback` in `Meta`: ❓ surprisingly works (callback called for `NewForm`)
```python3
class BaseForm(forms.ModelForm):
    class Meta:
        model = Person
        fields = ["id"]
        formfield_callback = callback

class NewForm(BaseForm):
    class Meta:
        model = Person
        fields = ["id", "name"]
```

So this is a mess right now 😄 By going your way (simplifying the fix for 24974) we make things clear but drop the support for my example number 2. Nevertheless, I'd prefer to go this way.